docs/proposals/metadata.md

```diff
@@ -0,0 +1,153 @@
+# Image Metadata
+
+## Abstract
+
+A proposal for the additional metadata for Docker images which will provide more
+context about the content of the image and define relations between Docker
+images.
+
+## Motivation
+
+Image metadata are needed to do automatic generation of the OpenShift v3
+resource templates, like a PodTemplates, ReplicationControllers, BuildConfigs
+and other.  Automatic generation of PodTemplate is also required for
+Deployments.
+
+## Constraints and Assumptions
+
+For the time being, there is no good way how to define metadata for Docker
+images other than specifying extra environment variables. There is an upstream
+work currently being done to fix this situlation and we should switch to that in
+the implementation in the future.
+
+We also assume that there might be more use-cases and metadata define for images
+that helps with generation stuff, this document only provides those needed in
+the short-term.
+
+## Use cases
+
+1. As an author of the Docker image, I want to specify a flag that will tell
+   whether the image I build can be used as an STI image builder, so that when
+   somebody use my image in OpenShift v3 the BuildConfig generated for this image
+   will default the build strategy to STI. For this, I only need to specify if my
+   entrypoint accepts a 'tar' stream that contains the application as an input.
+
+2. As an author of the Docker image, I want to specify list of services that the
+   image I'm building will provide, so in OpenShift v3 we can automatically
+   setup Routes, Services and health-checks for this image. For example, in case
+   I'm building 'mysql' image, I can specify `PUBLISHES=mysql/tcp/3128` so when
+   the image is deployed, we automatically setup a Service for it, add health-check
+   for 'mysql' service type and wire the Route for the 3128 port.
+
+3. As an user of OpenShift v3 I want to be able to say what services my
+   application will consume, so the platform can find the corresponding Docker
+   images and make the services available for my application.
+   For example, as an user I can say I want to have 'mysql' service available in
+   my application, so the platform will check the ImageRepositories for the
+   Image that have `PUBLISHES="mysql/tcp/3128"` and add it to my deployment.
+
+4. As an author of the Docker image, I want to specify what _secrets_ the image
+   I'm building require, so I can consume them in the Image.
+   For example I can specify `SECRETS=api/twitter` that should tell the platform
+   to lookup for the 'api/twitter' secret and make it available inside the
+   Image.
+
+5. As an author of the Docker image, I need to have a way to disable
+   'clustering' for my Image, in case the Docker image does not support this.
+   In other words, if the Docker image specify this, the ReplicationController
+   generated for this Image should never get `replicas` set higher than 1.
+
+## The list of proposed metadata
+
+### ACCEPT_TAR_STREAM
```

Comment:
trello card here: https://trello.com/c/rslmL9CM/498-3-sti-support-for-images-that-accept-tar-build-future